Source/WebCore/loader/ImageLoader.cpp

```diff
@@ -349,7 +349,10 @@ void ImageLoader::didUpdateCachedImage(RelevantMutation relevantMutation, Cached
 void ImageLoader::updateFromElementIgnoringPreviousError(RelevantMutation relevantMutation)
 {
     clearFailedLoadURL();
-    updateFromElement(relevantMutation);
+    Ref document = element().document();
+    document->eventLoop().queueTask(TaskSource::InternalAsyncTask, [imageLoader = this, relevantMutation] {
```

Comment:
This should probably be queueMicrotask()?

Reply:
yeah, for sure!

Reply:
I have to read through the associated element code but is there denouncing somewhere?

Reply:
What do you mean by debouncing in this context?

Reply:
What I mean is that this method is called repeatedly each time an attribute changes:
https://github.com/WebKit/WebKit/blob/ed73610d6dc93d5e5b5065b34dabd11c878dcbf0/Source/WebCore/html/HTMLImageElement.cpp#L412

which means if you do:

img.src = "..."
img.referrerPolicy = "..."
img.crossOrigin = "..."

your code makes it queue 3 microtasks.

Maybe that's fine since the old code always ran updateFromElement() 3 times in that case too, but I think the spec intends only a single microtask in those situations.

Reply:
I believe this is no longer the case. PTAL?